server/scheduler.go

```diff
@@ -297,21 +297,22 @@ func scheduleRemovePeer(cluster *clusterInfo, s Selector, filters ...Filter) (*r
 
 // scheduleTransferLeader schedules a region to transfer leader to the peer.
 func scheduleTransferLeader(cluster *clusterInfo, s Selector, filters ...Filter) (*regionInfo, *metapb.Peer) {
-	sourceStores := cluster.getStores()
-
-	source := s.SelectSource(sourceStores, filters...)
-	if source == nil {
+	region := cluster.randomRegion()
+	if region == nil {
 		return nil, nil
 	}
 
-	region := cluster.randLeaderRegion(source.GetId())
-	if region == nil {
+	leaderStore := cluster.getStore(region.Leader.GetStoreId())
```

Comment:
random region let hot leader store get less chance to transfer leader. I see the `maxScheduleRetries` just 10. if region large than 100000, it just sampling 0.01 percent, is it seems very possibly gets nil option or transfer between free machine? I think we also need adjust schedule-retrys with the regions total number and schedule-interval increase way.